src/canvas.py

```diff
@@ -44,29 +44,28 @@ def __init__(self, application):
         self.right_axis = self.axis.twinx()
         self.top_left_axis = self.axis.twiny()
         self.top_right_axis = self.top_left_axis.twinx()
-        self.set_ticks()
-        color_rgba = \
-            utilities.lookup_color(self.props.application, "accent_color")
+        self._axes = [
+            self.axis, self.top_left_axis,
+            self.right_axis, self.top_right_axis,
+        ]
+        color_rgba = self.get_style_context().lookup_color("accent_color")[1]
         self.rubberband_edge_color = utilities.rgba_to_tuple(color_rgba, True)
         color_rgba.alpha = 0.3
         self.rubberband_fill_color = utilities.rgba_to_tuple(color_rgba, True)
-        self.legends = []
-        for axis in [self.right_axis, self.top_left_axis,
-                     self.top_right_axis]:
-            axis.get_xaxis().set_visible(False)
-            axis.get_yaxis().set_visible(False)
         DummyToolbar(self)
         self.highlight = Highlight(self)
-
-        property_ignorelist = [
-            "legend", "legend_position", "use_custom_style", "custom_style",
-        ]
+        self._legend = True
+        self._legend_position = LEGEND_POSITIONS[0]
+        self._handles = []
 
         for prop in dir(self.props.application.props.figure_settings.props):
-            if prop not in property_ignorelist:
+            if prop not in ["use_custom_style", "custom_style"]:
                 self.props.application.props.figure_settings.bind_property(
-                    prop, self, prop, 2,
+                    prop, self, prop, 1 | 2,
```

Comment:
I know this was already in the old code, but I tried to look a bit into the bind_property function from GObject, where this seems to come from.  Binding seems to be

`  g_object_bind_property (action, "active", widget, "sensitive", 0);  `

But the docs didn't make it very obvious for me what each argument does. Mainly what does the 1 | 2 pipe do for instance? 

Not relevant for the review, just wanted to know for my own understanding.